src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -54,15 +54,29 @@ export function DiscoverMainRoute({ services, history }: DiscoverMainProps) {
   const [indexPatternList, setIndexPatternList] = useState<
     Array<SavedObject<IndexPatternAttributes>>
   >([]);
-
+  const [isNewKibanaInstance, setNewKibanaInstance] = useState(false);
   const { id } = useParams<DiscoverLandingParams>();
 
+  useEffect(() => {
+    const fetchIsNewKibanaInstance = async () => {
+      const hasUserIndexPattern = await data.indexPatterns.hasUserDataView().catch(() => true);
+
+      setNewKibanaInstance(!hasUserIndexPattern);
+    };
+
+    fetchIsNewKibanaInstance();
+  }, [data.indexPatterns]);
+
   useEffect(() => {
     const savedSearchId = id;
 
     async function loadDefaultOrCurrentIndexPattern(searchSource: ISearchSource) {
       try {
-        await data.indexPatterns.ensureDefaultDataView();
+        const defaultIndexPattern = await data.dataViews.getDefaultDataView();
+        if (!defaultIndexPattern) {
+          setNewKibanaInstance(true);
+          return;
+        }
```

Comment:
Wouldn't it make it simpler to check also here for `data.indexPatterns.hasUserDataView` and instead of updating new state to use `core.application.navigateToApp('kibanaOverview', { path: '#' });? 
Also it would prevent the unlikely case that `data.indexPatterns.hasUserDataView` has a delay and returned false, while savedSearch and You-know-who (we need to update our code to remove all those namings) already returned providing truthy values. This would start rendering the UI of the page, and suddenly redirect away ... Don't think this can happen, but anyway, we could prevent it by simplifying code.

Reply:
I wouldn't change this to `hasUserDataView`, cause I believe this check is here with a reason (if I understand things correctly). We want to ensure that there is a default dataview. The only difference is that the method on `dataViews` now can return a null value, where the one on `indexPatterns` didn't.

I think this should still redirect to data views management page, this was an oversight on my part. If I'm not wrong, this would have hit this code path before:
https://github.com/elastic/kibana/pull/123366/files#diff-ba3b80e85abb5cac62f35f7b5819a668f70c8e7ea7c135e3dad00b1dbec87417L114

If this method is now not throwing an exception, we should redirect here.

Reply:
I wouldn't change it, I would add the `hasUserDataView` here, 
```
const defaultDataView = await data.dataViews.getDefaultDataView();
const hasUserDataView = await data.dataViews.hasUserDataView().catch(() => true);
if(!defaultDataView || !hasUserDataView) {
   ... redirectCode...
}
```
So no more state would be needed